pkg/router/controller/router_controller.go

```diff
@@ -304,6 +313,10 @@ func convertEndpointSliceToEndpointSubset(items []discoveryv1beta1.EndpointSlice
 			}
 		}
 
+		if len(addresses) == 0 {
```

Comment:
Is this just an optimization missed in prior endpointslice work?

Reply:
If this is idle specific, maybe a comment would be helpful?

Reply:
I thought I had commented on this. IIRC (and my memory is fuzzy since I last looked at this) this is/was absolutely essential.  Will debug and add a comment.

Reply:
Marking as resolved as we're now checking the service for the idle annotation instead and the logic has moved to `subsetHasAddresses`.